### PR TITLE
Restructuring and addition to permissions docs

### PR DESCRIPTION
**Pull request by xingluw**
Restructuring and adding of material and examples to permissions/grants documentation. The goal is to make the permissions concept easier to understand and implement for users.

### CODE REVIEW

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 1]
```diff
@@ -10,69 +10,43 @@ description: |-
...
-attempts to marry flexibility with usability. This page discusses the permission
-model's fundamental concepts, provides examples of the specific forms of allowed
-grants, and contains a table that acts as an easy cheat sheet to help those new
-to its grant syntax with crafting roles.
+attempts to marry flexibility with usability. This section discusses the permission
+model's fundamental concepts, provides examples grant strings, and contains a table 
+that acts as an cheat sheet to help those new to its grant syntax with crafting roles.
 
-Boundary's [domain model](/boundary/docs/concepts/domain-model) is based on resource
+Boundary's [domain model](/docs/concepts/domain-model) is based on resource
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/index.mdx:17]:
```suggestion
Boundary's [domain model](/boundary/docs/concepts/domain-model) is based on resource
```

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 2]
```diff
@@ -10,69 +10,43 @@ description: |-
...
-the permissions that should be granted on them.
-
-There are currently two types of selectors:
 
-- An `id` field that indicates a specific resource or a wildcard to match all
+## Grant Strings
 
-- A `type` field that indicates a specific resource type or a wildcard to match
-  all; this might also be used to grant permissions on collections of resources
+A [grant string](/docs/concepts/security/permissions/grants) is a mapping
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/index.mdx:29]:
```suggestion
A [grant string](/boundary/docs/concepts/security/permissions/grants) is a mapping 
```

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 3]
```diff
@@ -10,69 +10,43 @@ description: |-
...
+granted on them. Grant strings can be composed of up to 4 parts: 
+- ID
+- Type
+- Actions 
+- Output Fields
 
-Selectors are used to indicate which resources on which the grant should apply,
-using specific IDs or wildcard IDs and type selectors. (The acceptable grant
-string formats are detailed later on this page.)
+To learn about writing grant strings with examples, see [Grant Strings](/docs/concepts/security/permissions/grants).
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/index.mdx:37]:
```suggestion
To learn about writing grant strings with examples, see [Grant Strings](/boundary/docs/concepts/security/permissions/grants).
```

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 4]
```diff
@@ -10,69 +10,43 @@ description: |-
...
-- An `output_fields` field indicating which top-level fields to return in the
-  response (0.2.1+)
-
-Grant strings can be supplied via a human-friendly string syntax or via JSON.
-
-### Roles
-
-Roles map grant strings to _principals_, currently users and groups. Every role
-assigns grants within a specific scope: either the scope in which the role
+A [role](/docs/concepts/domain-model/roles) maps grant strings to principals (users and groups).
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/index.mdx:41]:
```suggestion
A [role](/boundary/docs/concepts/domain-model/roles) maps grant strings to principals (users and groups). 
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+- An `actions` field indicating which actions to allow the client to perform on
+  the resources matched by `id` and `type`
+
+- An `output_fields` field indicating which top-level fields to return in the
+  response
+
+The format of a full grant string is:
+
+- &#32 `id=<id>;type=<type>;actions=<action-list>;output_fields=<fields-list>`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:46]:
```suggestion
- &#32; `id=<id>;type=<type>;actions=<action-list>;output_fields=<fields-list>`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+The **ID** is the specific resource that you select for the grant.
+You can use a [wildcard](/docs/concepts/security/permissions/advanced#wildcard) (*) to include all resources of the type and scope.
+Each grant string supports only 1 ID or a wildcard. By specifying a single ID, the top-level resource 
+is "pinned", meaning the permissions will only apply to that resource and subordinate resources.
+
+The examples of valid IDs are:
+
+- &#32 `id=*` (all)
+- &#32 `id=ttcp_123456789`
+- &#32 `id=w_abcdefghi`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:65]:
```suggestion
- &#32; `id=*` (all)
- &#32; `id=ttcp_123456789`
- &#32; `id=w_abcdefghi`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+include all resource types in the scope.
+Each grant string supports only 1 type or a wildcard.
+
+Some commonly used types are:
+
+- &#32 `*` (all)
+- &#32 `session`
+- &#32 `target`
+- &#32 `user`
+- &#32 `worker`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:81]:
```suggestion
- &#32; `*` (all)
- &#32; `session`
- &#32; `target`
- &#32; `user`
- &#32; `worker`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+You can also use a [wildcard](/boundary/docs/concepts/security/permissions/permission-grant-formats#wildcard-id) to include all valid actions.
+Each grant string can include one or more comma separated actions.
+Some commonly used actions include:
+
+- &#32 `*` (all)
+- &#32 `list`
+- &#32 `create`
+- &#32 `read`
+- &#32 `update`
+- &#32 `delete`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:98]:
```suggestion
- &#32; `*` (all)
- &#32; `list`
- &#32; `create`
- &#32; `read`
- &#32; `update`
- &#32; `delete`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+**Output fields** are top-level fields that are returned in the response. Output fields
+are **optional** and not commonly used.
+
+Examples of valid output fields include:
+
+- &#32 `none`
+- &#32 `id`
+- &#32 `name`
+- &#32 `description`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:113]:
```suggestion
- &#32; `none`
- &#32; `id`
- &#32; `name`
- &#32; `description`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+## Common Grant Examples
+
+The following are examples of commonly used grant strings for specific roles and scopes.
+
+**All Actions** 
+
+This grant string allows users to perform any actions on any resources in the scope:
+
+- &#32 `id=*;type=*;actions=*`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:125]:
```suggestion
- &#32; `id=*;type=*;actions=*`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+
+**View All** 
+
+This grant string allows users to list and read any resources in the scope:
+
+- &#32 `id=*;type=*;actions=read,list`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:131]:
```suggestion
- &#32; `id=*;type=*;actions=read,list`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+**View Targets Only** 
+
+This **project scope** grant string allows users to list and read any targets in the specific project.
+This example is only applicable to project scopes because targets can only exist inside a project:
+
+- &#32 `id=*;type=target;actions=read,list`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:138]:
```suggestion
- &#32; `id=*;type=target;actions=read,list`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+
+**Connect To Any Target**
+
+This **project scope** grant string allows users to list, read, and connect to any targets in a specific project.
+In addition, these grants allow the user to list, read, and cancel any sessions initiated by the same user.
+This example is only applicable to project scopes, since targets can only exist in a project:
+
+- &#32 `id=*;type=target;actions=list,read,authorize-session`
+- &#32 `id=*;type=session;actions=read:self,cancel:self,list`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:147]:
```suggestion
- &#32; `id=*;type=target;actions=list,read,authorize-session`
- &#32; `id=*;type=session;actions=read:self,cancel:self,list`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+**Connect To A Specific Target** 
+
+This **project scope** grant string allows users to list, read, and connect to a specific target.
+In addition, these grants let users list, read, and cancel any sessions initiated by the same user.
+If you want to include several specific targets, you can duplicate the first grant string using the respective IDs of those targets.
+This example is only applicable to project scopes, since targets can only exist in a project:
+
+- &#32 `id=<target-id>;actions=read,authorize-session`
+- &#32 `type=target;actions=list`
+- &#32 `id=*;type=session;actions=read:self,cancel:self,list`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:158]:
```suggestion
- &#32; `id=<target-id>;actions=read,authorize-session`
- &#32; `type=target;actions=list`
- &#32; `id=*;type=session;actions=read:self,cancel:self,list`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+However, you are still able to use this set of grants to connect to a target using the CLI.
+</Note>
+
+## General Grant Examples
+
+The following are examples of general grant strings for any scope:
+
+**Manage All Of 1 Resource Type**
+
+- &#32 `id=*;type=<resource-type>;actions=*`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:172]:
```suggestion
- &#32; `id=*;type=<resource-type>;actions=*`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+In addition, these grants let users list, read, and cancel any sessions initiated by the same user.
+If you want to include several specific targets, you can duplicate the first grant string using the respective IDs of those targets.
+This example is only applicable to project scopes, since targets can only exist in a project:
+
+- &#32 `id=<target-id>;actions=read,authorize-session`
+- &#32 `type=target;actions=list`
+- &#32 `id=*;type=session;actions=read:self,cancel:self,list`
+
+<Note>
+There is a known issue that affects the [Project] Connect To A Specific Target grant strings.
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:161]: Isn't this known issue fixed in 0.13.0? Maybe we can remove this.

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+
+The following are examples of general grant strings for any scope:
+
+**Manage All Of 1 Resource Type**
+
+- &#32 `id=*;type=<resource-type>;actions=*`
+
+**View All Of 1 Resource Type**
+
+- &#32 `id=*;type=<resource-type>;actions=list,read`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:176]:
```suggestion
- &#32; `id=*;type=<resource-type>;actions=list,read`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+
+**Manage A Specific Resource IItem**
+
+- &#32 `type=<resource-type>;actions=list`
+- &#32 `id=<resource-id>;actions=*`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:181]:
```suggestion
- &#32; `type=<resource-type>;actions=list`
- &#32; `id=<resource-id>;actions=*`
```

File: website/content/docs/concepts/security/permissions/grants.mdx [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+
+**View A Specific Resource Item**
+
+- &#32 `type=<resource-type>;actions=list`
+- &#32 `id=<resource-id>;actions=read`
```
Comment (Dan-Heath) [inline website/content/docs/concepts/security/permissions/grants.mdx:186]:
```suggestion
- &#32; `type=<resource-type>;actions=list`
- &#32; `id=<resource-id>;actions=read`
```

**Review by Dan-Heath — CHANGES_REQUESTED** [review]
@xingluw I am sorry I gave you some bad advice regarding the links and the blank space entities. I think the suggestions in this review should clear up the content errors on this PR and hopefully get the Vercel build going.

I will review again for content.

File: website/content/docs/concepts/security/permissions/grants.mdx (hunk 16 above)
Comment (xingluw) [reply website/content/docs/concepts/security/permissions/grants.mdx:161]: I need to validate this first, will do soon

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 20]
```diff
@@ -88,4 +62,6 @@ scope ID and one or more of the following are true:
   role
 
 Roles are composable; a user's final set of grants will be composed of grants
-that originate from all matching roles.
\ No newline at end of file
+that originate from all matching roles.
+
+To learn more about Boundary's access management system, see [IAM](/docs/concepts/security/iam).
```
Comment (xingluw) [inline website/content/docs/concepts/security/permissions/index.mdx:67]:
```suggestion
To learn more about Boundary's access management system, see IAM.
```

File: website/content/docs/concepts/security/permissions/index.mdx [hunk 21]
```diff
@@ -88,4 +62,6 @@ scope ID and one or more of the following are true:
   role
 
 Roles are composable; a user's final set of grants will be composed of grants
-that originate from all matching roles.
\ No newline at end of file
+that originate from all matching roles.
+
+To learn more about Boundary's access management system, see IAM.
```
Comment (xingluw) [inline website/content/docs/concepts/security/permissions/index.mdx:67]:
```suggestion
To learn more about Boundary's access management system, see IAM [LINK TBD].
```

**Comment by xingluw** [comment]
making new PR due to issues with vercel

**Pull request closed without merging**